dashboard/modules/stats_collector/stats_collector_head.py

```diff
@@ -0,0 +1,155 @@
+        hostname = req.match_info.get("hostname")
+        node_info = await DataOrganizer.get_node_info(hostname)
+        return await dashboard_utils.rest_response(
+            success=True, message="Node detail fetched.", detail=node_info)
+
+    async def _update_actors(self):
+        # Subscribe actor channel.
+        aioredis_client = self._dashboard_head.aioredis_client
+        receiver = Receiver()
+
+        key = "{}:*".format(stats_collector_consts.ACTOR_CHANNEL)
+        pattern = receiver.pattern(key)
+        await aioredis_client.psubscribe(pattern)
+        logger.info("Subscribed to %s", key)
+
+        # Get all actor info.
+        while True:
+            try:
+                logger.info("Getting all actor info from GCS.")
+                request = gcs_service_pb2.GetAllActorInfoRequest()
+                reply = await self._gcs_actor_info_stub.GetAllActorInfo(
+                    request, timeout=2)
+                if reply.status.code == 0:
+                    result = {}
+                    for actor_info in reply.actor_table_data:
+                        result[binary_to_hex(actor_info.actor_id)] = \
+                            actor_table_data_to_dict(actor_info)
+                    DataSource.actors.reset(result)
+                    logger.info("Received %d actor info from GCS.",
+                                len(result))
+                    break
+                else:
+                    raise Exception(
+                        f"Failed to GetAllActorInfo: {reply.status.message}")
+            except Exception as ex:
+                logger.exception(ex)
+                await asyncio.sleep(stats_collector_consts.
+                                    RETRY_GET_ALL_ACTOR_INFO_INTERVAL_SECONDS)
+
+        # Receive actors from channel.
+        async for sender, msg in receiver.iter():
+            try:
+                _, data = msg
+                pubsub_message = ray.gcs_utils.PubSubMessage.FromString(data)
+                actor_info = ray.gcs_utils.ActorTableData.FromString(
+                    pubsub_message.data)
+                DataSource.actors[binary_to_hex(actor_info.actor_id)] = \
+                    actor_table_data_to_dict(actor_info)
+            except Exception as ex:
+                logger.exception(ex)
+
+    @async_loop_forever(
+        stats_collector_consts.NODE_STATS_UPDATE_INTERVAL_SECONDS)
+    async def _update_node_stats(self):
+        for ip, stub in self._stubs.items():
+            node_info = DataSource.nodes.get(ip)
+            if node_info["state"] != "ALIVE":
+                continue
+            try:
+                reply = await stub.GetNodeStats(
+                    node_manager_pb2.GetNodeStatsRequest(), timeout=2)
+                reply_dict = node_stats_to_dict(reply)
+                DataSource.node_stats[ip] = reply_dict
+            except Exception as ex:
+                logger.exception(ex)
```

Comment:
Could you add a bit more information to the error message here? logger.exception will already capture the exception in its body so you could do 
```
except Exception:
    logger.exception(f"Error fetching node stats for node with ip={ip}")
```
https://docs.python.org/3/library/logging.html#logging.Logger.exception

Reply:
Good suggestion. `logger.exception(ex)` is to show the message of `ex` with traceback. I will refine the message with details.